umh-core/pkg/config/ctxrwmutex/ctxrwmutex.go

```diff
@@ -0,0 +1,42 @@
+package ctxrwmutex
+
+import (
+	"context"
+
+	"golang.org/x/sync/semaphore"
+)
+
+// CtxRWMutex is a context aware RWMutex
```

Comment:
maybe add a comment here that the only reason we are using a weighted semaphore is because of the context.